### DISCUSSION TITLE
AssertionError: One of self.n_classes or n_classes must be an integer

### DISCUSSION THREAD

**Discussion by Yangtting** [Project-MONAI/tutorials#485] [Q&A]
I used "spleen_segmentation_3d.ipynb"on my dataset. 
My dataset included images and labels (0:background, 1:aortic root, 2:Plaque)
I changed model as
model = UNet(
    dimensions=3,
    in_channels=1,
    out_channels=3,
    channels=(16, 32, 64, 128, 256),
    strides=(2, 2, 2, 2),
    num_res_units=2,
    norm=Norm.BATCH,
).to(device)
And change 
"
post_pred = Compose([EnsureType(), AsDiscrete(argmax=True, to_onehot=3)])
post_label = Compose([EnsureType(), AsDiscrete(to_onehot=3)])
"
When I run code, it occured bug in valiadation process.
"AssertionError: One of self.n_classes or n_classes must be an integer"
"RuntimeError: applying transform <monai.transforms.post.array.AsDiscrete object at 0x000001EE6C51EF70>"

**Comment by Nic-Ma** [comment]
Hi @Yangtting ,

Thanks for your interest here.
May I know your MONAI version in the test? Please update to run with MONAI v0.8.0 again.

Thanks.

**Reply by Yangtting** [reply]
Oh, thank you very much! 
I used MONAI v0.6.0 before, it works great with v0.8.0!

**Reply by Nic-Ma** [reply]
The root cause is that you are using the new style API but the package is old.

Thanks.
